views/js/runner/ui/toolbox/toolbox.js

```diff
@@ -0,0 +1,213 @@
+        createEntry: function createEntry(config)  {
+            var item = entryFactory().init(config);
+            this.allItems.push(item);
+            return item;
+        },
+
+        /**
+         * Create a text component instance
+         * @param {Object} config
+         * @param {String} config.control - will be used as the instance id and in the data-control html attribute
+         * @param {String} config.text - the text content
+         * @param {String} config.className - an extra class
+         * @returns {Component} the create instance
+         */
+        createText: function createText(config) {
+            var text = textFactory().init(config);
+            this.allItems.push(text);
+            return text;
+        },
+
+        /**
+         * If the given item belongs to a menu
+         * @param {Object} item
+         * @returns {Boolean}
+         */
+        hasMenu: function hasMenu(item) {
+            return item && _.isFunction(item.getMenuId) && item.getMenuId();
+        }
+    };
+
+
+    /**
+     * Default renderer. It simply appends all the registered items in the toolbox, one after the other
+     * @param {jQuery} $container - where to render
+     */
+    function defaultRenderer($container) {
+        var self = this,
+            menuEntries = [];
+
+        // render first level
+        if (_.isArray(this.allItems)) {
+            this.allItems.forEach(function (item) {
+                // items belonging to menus will be processed later
+                if (! self.hasMenu(item)) {
+                    item.render($container);
+                }
+            });
+        }
+
+        // Render each menu
+        this.allMenus.forEach(function (menu) {
+            var menuId = menu.getId();
+
+            // first, we gather all items relevant to the current menu
+            menuEntries = self.allItems.filter(function (item) {
+                return (item.getType() === 'entry' && item.getMenuId() === menuId);
+            });
+
+            // we then add entries to the current menu
+            menuEntries.forEach(function (item) {
+                menu.addItem(item);
+            });
+
+            // and finally render the whole menu
+            menu.renderItems();
+        });
+    }
+
+    /**
+     * The toolbox factory
+     */
+    return function toolbarComponentFactory(specs, defaults) {
+        var toolbarComponent;
+
+        specs = _.defaults(specs || {}, toolbarComponentApi);
+
+        toolbarComponent = componentFactory(specs, defaults)
+            .on('init', function () {
+                this.initToolbox();
+            })
+            // overridable renderer
+            .on('render.defaultRenderer', defaultRenderer)
+
+            // non-overridable renderer
+            .on('render', function() {
+                var self = this;
+
+                // fixme: try to bind this behavior on the blur event of each menu
+                $(document).off('.toolboxmenu');
```

Comment:
multiple instanciation of jquery ... should use var $doc = $(document) instead, no ?